File: scala/prime-factors/src/main/scala/Primefactors.scala
import scala.collection.mutable.ListBuffer

object PrimeFactors {
  def factors(i: Long) = {

    val factors = ListBuffer[Int]()
    var dup = i
    var div = 2

    while (dup > 1){
      while (dup % div == 0){
        factors += div
        dup /= div
      }
      div += 1
    }


    factors.toList

  }

}

File: r/prime-factors/prime-factors.R
prime_factors = function(n) {
  factors = c()
  cur_prime = 2

  while (n > 1) {
    if (n %% cur_prime == 0) {
      n = n / cur_prime
      factors[length(factors) + 1] = cur_prime
    } else {
      cur_prime = cur_prime + 1
    }
  }
  factors
}
